src/odb/src/db/dbAccessPoint.cpp

```diff
@@ -0,0 +1,308 @@
+
+  // User Code Begin Out
+  // User Code End Out
+  DIFF_END
+}
+_dbAccessPoint::_dbAccessPoint(_dbDatabase* db)
+{
+  // User Code Begin Constructor
+  accesses_.resize(6, false);
+  // User Code End Constructor
+}
+_dbAccessPoint::_dbAccessPoint(_dbDatabase* db, const _dbAccessPoint& r)
+{
+  layer_ = r.layer_;
+  type_low_ = r.type_low_;
+  type_high_ = r.type_high_;
+  // User Code Begin CopyConstructor
+  // User Code End CopyConstructor
+}
+
+dbIStream& operator>>(dbIStream& stream, _dbAccessPoint& obj)
+{
+  stream >> obj.point_;
+  stream >> obj.layer_;
+  stream >> obj.accesses_;
+  stream >> obj.type_low_;
+  stream >> obj.type_high_;
+  // User Code Begin >>
+  // User Code End >>
+  return stream;
+}
+dbOStream& operator<<(dbOStream& stream, const _dbAccessPoint& obj)
+{
+  stream << obj.point_;
+  stream << obj.layer_;
+  stream << obj.accesses_;
+  stream << obj.type_low_;
+  stream << obj.type_high_;
+  // User Code Begin <<
+  // User Code End <<
+  return stream;
+}
+
+_dbAccessPoint::~_dbAccessPoint()
+{
+  // User Code Begin Destructor
+  // User Code End Destructor
+}
+
+// User Code Begin PrivateMethods
+// User Code End PrivateMethods
+
+////////////////////////////////////////////////////////////////////
+//
+// dbAccessPoint - Methods
+//
+////////////////////////////////////////////////////////////////////
+
+void dbAccessPoint::setPoint(Point point)
+{
+  _dbAccessPoint* obj = (_dbAccessPoint*) this;
+
+  obj->point_ = point;
+}
+
+Point dbAccessPoint::getPoint() const
+{
+  _dbAccessPoint* obj = (_dbAccessPoint*) this;
+  return obj->point_;
+}
+
+void dbAccessPoint::setLayer(dbTechLayer* layer)
+{
+  _dbAccessPoint* obj = (_dbAccessPoint*) this;
+
+  obj->layer_ = layer->getImpl()->getOID();
+}
+
+void dbAccessPoint::setAccesses(std::vector<bool> accesses)
+{
+  _dbAccessPoint* obj = (_dbAccessPoint*) this;
+
+  obj->accesses_ = accesses;
+}
+
+void dbAccessPoint::getAccesses(std::vector<bool>& tbl) const
+{
+  _dbAccessPoint* obj = (_dbAccessPoint*) this;
+  tbl = obj->accesses_;
+}
+
+// User Code Begin dbAccessPointPublicMethods
+
+void dbAccessPoint::setTypeLow(AccessType type_low)
+{
+  _dbAccessPoint* obj = (_dbAccessPoint*) this;
+
+  obj->type_low_ = type_low;
+}
+
+dbAccessPoint::AccessType dbAccessPoint::getTypeLow() const
+{
+  _dbAccessPoint* obj = (_dbAccessPoint*) this;
+  return dbAccessPoint::AccessType(obj->type_low_);
+}
+
+void dbAccessPoint::setTypeHigh(AccessType type_high)
+{
+  _dbAccessPoint* obj = (_dbAccessPoint*) this;
+
+  obj->type_high_ = type_high;
+}
+
+dbAccessPoint::AccessType dbAccessPoint::getTypeHigh() const
+{
+  _dbAccessPoint* obj = (_dbAccessPoint*) this;
+  return dbAccessPoint::AccessType(obj->type_high_);
+}
+
+void dbAccessPoint::setAccess(bool access, dbDirection dir)
+{
+  // 0 = E, 1 = S, 2 = W, 3 = N, 4 = U, 5 = D
+  _dbAccessPoint* obj = (_dbAccessPoint*) this;
+  switch (dir) {
+    case dbDirection::EAST:
+      obj->accesses_[0] = access;
```

Comment:
I wanted to match the structure in TritonRoute so that I can set accesses by passing the bool vector directly. if this is inconvenient, I can change the index to be the dir and set the accesses direction by direction.